addons/mail/static/src/models/thread/thread.js

```diff
@@ -720,6 +720,16 @@ function factory(dependencies) {
             this.fetchAndUpdateSuggestedRecipients();
         }
 
+        /**
+         * Returns the name of the given partner in the context of this thread.
+         *
+         * @param {mail.partner} partner
+         * @returns {string}
+         */
+        getMemberName(partner) {
+            return partner.nameOrDisplayName;
+        }
+
```

Comment:
It's okay as a method for the time being.

If this was fully and correctly integrated in model, this should be a relational field to a record that represents contextual info of this partner in this thread. It'd basically be a related without `im_livechat` installed, and a computed field for handling `livechat_name` with `im_livechat` module installed on livechat channels.

Note that it's cumbersome to make a model containing contextual info of partner being member of channel... Because it's composition... Would be nicer if we had a way to easily define "contextual fields" on a record, so it's not a different record and would just be something akin to `thread.members[i].threadName` (ignore the syntax, just the idea).